packages/web3-eth/src/errors.ts

```diff
@@ -1,21 +1,159 @@
 /* eslint-disable max-classes-per-file */
 
-import { Web3Error } from 'web3-utils';
+import { Numbers, Web3Error } from 'web3-utils';
 
 export class InvalidTransactionWithSender extends Web3Error {
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	public constructor(value: any) {
-		// TODO Discuss this naive approach to logging object
-		// Does not account for non JSON properties
+	public constructor(value: unknown) {
 		super(JSON.stringify(value), 'invalid transaction with sender');
 	}
 }
 
 export class InvalidTransactionCall extends Web3Error {
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	public constructor(value: any) {
-		// TODO Discuss this naive approach to logging object
-		// Does not account for non JSON properties
+	public constructor(value: unknown) {
 		super(JSON.stringify(value), 'invalid transaction call');
 	}
 }
+
+export class MissingCustomChainError extends Web3Error {
+	public constructor() {
+		// TODO - Discuss what to pass as value
+		super('N/A', 'If tx.common is provided it must have tx.common.customChain');
+	}
+}
+
+export class MissingCustomChainIdError extends Web3Error {
+	public constructor() {
+		super(
+			// TODO - Discuss what to pass as value
+			'N/A',
+			'If tx.common is provided it must have tx.common.customChain and tx.common.customChain.chainId',
```

Comment:
`Web3Error` requires a string value to be passed to be included in the error message. For this error it doesn't make sense to pass a value in and I wasn't sure what to pass to `Web3Error`'s constructor

Reply:
instead of 'N/A' may be good to pass empty string

Reply:
Would suggest to pass name of the error constructor `MissingCustomChainIdError`.

Reply:
I think including the error constructor is the better idea. I like being able to use `instanceOf` for each specific error